prov/rxm/src/rxm_ep.c

```diff
@@ -1284,23 +1311,22 @@ rxm_ep_send_common(struct rxm_ep *rxm_ep, const struct iovec *iov, void **desc,
 		return ret;
 	} else {
 		assert(!(flags & FI_INJECT));
-		if ((data_len <= rxm_ep->sar_limit) &&
-		    /* Roughly estimate whether all SAR segments can be sent w/o
-		     * retransmission due to lack of space in the TX queue */
-		    (rxm_ep_sar_estimate_segments_num(rxm_ep, data_len) <=
-						rxm_ep->msg_info->tx_attr->size)) {
-			return rxm_ep_sar_tx_send(rxm_ep, rxm_conn, context, count, iov,
-						  data_len, data, flags, comp_flags, tag, op);
-		} else {
-			ret = rxm_ep_alloc_lmt_tx_res(rxm_ep, rxm_conn, context,
-						      (uint8_t)count, iov, desc,
-						      data_len, data, flags, comp_flags,
-						      tag, op, &tx_entry);
-			if (OFI_UNLIKELY(ret < 0))
-				return ret;
-			return rxm_ep_lmt_tx_send(rxm_ep, rxm_conn, tx_entry,
-						  sizeof(struct rxm_pkt) + ret);
+		if (data_len <= rxm_ep->sar.limit) {
+			size_t segs_cnt = rxm_ep_sar_calc_segs_cnt(rxm_ep, data_len);
+
+			if (segs_cnt <= rxm_ep->sar.segs_cnt_limit) {
```

Comment:
Do we need this check? As long as `data_len <= rxm_ep->sar.limit`, `segs_cnt` should be less than `rxm_ep->sar.segs_cnt_limit`

Reply:
I'm going to replace the
```
if (segs_cnt <= rxm_ep->sar.segs_cnt_limit) {
         ...
}
```
with something like
```
if (segs_cnt <= rxm_conn->remain_tx_size) {
         ...
}
```

`rxm_conn->remain_tx_size` will be initiated by `rxm_ep->msg_info->tx_attr->size` and each TX will decrement this value. When the TX operation is completed, the value is incremented

Reply:
I removed this branch